marlowe-spec-test/src/Marlowe/Spec/Core/Semantics.hs

```diff
@@ -45,10 +111,238 @@ divisionRoundsTowardsZeroTest interpret = reproducibleProperty "Division roundin
     numerator <- run $ generateT $ genValue interpret
     denominator <- run $ generateT
         (genValue interpret
-          `suchThat` (\d -> (abs_int $ evalValue env state numerator) `less_int` (abs_int $ evalValue env state d))
+          `suchThat` (\d -> (Arith.abs_int $ evalValue env state numerator) `Arith.less_int` (Arith.abs_int $ evalValue env state d))
         )
     let
         req :: Request JSON.Value
         req = EvalValue env state (DivValue numerator denominator)
         successResponse = RequestResponse $ toJSON (0 :: Int)
-    assertResponse interpret req successResponse
\ No newline at end of file
+    assertResponse interpret req successResponse
+
+fixIntervalTest :: InterpretJsonRequest -> TestTree
+fixIntervalTest interpret = reproducibleProperty "Calling fixInterval test" do
+    i@(Arith.Int_of_integer a, Arith.Int_of_integer b) <- run $ generate $ genInterval
+    state <- run $ generateT $ genState interpret
+    let
+        req :: Request JSON.Value
+        req = FixInterval (a, b) state
+        successResponse = RequestResponse $ toJSON $ fixInterval i state
+    assertResponse interpret req successResponse
+
+computeTransactionTest :: InterpretJsonRequest -> TestTree
+computeTransactionTest interpret = reproducibleProperty "Calling computeTransaction test" do
+    state <- run $ generateT $ genState interpret
+    contract <- run $ generateT $ genContract interpret
+    transaction <- run $ generateT $ genTransaction interpret
```

Comment:
This doesn't seem to be using the SemiArbitrary instances.

Reply:
Ok, change to use the `SemiArbitrary` instance as well